### ISSUE TITLE
Can't run server after cloning

### ISSUE THREAD

**Issue by TAnas0** [tookit/vue-material-admin#8]
Hey,

I tried following the instructions in the Readme. I cloned the repository and ran `npm install`.
I resolved some unmet dependencies that `npm` warned me about (not sure if they are related to the project) using:
`npm install less@^3.0.0`
`npm install ajv@^6.0.0`

When trying to run the server using `npm run dev`, I get the following error:

```
Failed to compile with 1 errors
This dependency was not found:

* @/components/widgets/card/NameCard in ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/pages/widgets/Social.vue

To install it, you can run: npm install --save @/components/widgets/card/NameCard
```

I would appreciate help on this issue.

Thanks.
#### Additional informations:
```bash
node -v
v10.6.0

npm -v
6.1.0
```

**Comment by AntoninSlejska** [comment]
Hi TAnas0,
I tested it and everything works, but I have older versions of node and npm. In the console it looks like:

```
$ node -v
v8.9.4

$ npm -v
5.6.0

$ git clone https://github.com/tookit/vue-material-admin.git vma
Cloning into 'vma'...
remote: Counting objects: 417, done.
remote: Compressing objects: 100% (365/365), done.
remote: Total 417 (delta 66), reused 386 (delta 40), pack-reused 0
Receiving objects: 100% (417/417), 15.02 MiB | 1.67 MiB/s, done.
Resolving deltas: 100% (66/66), done.

$ cd vma

$ npm install

> chromedriver@2.38.3 install C:\Workspace\Sites\vma\node_modules\chromedriver
> node install.js

Downloading https://chromedriver.storage.googleapis.com/2.38/chromedriver_win32.zip
Saving to C:\Users\Slejska\AppData\Local\Temp\chromedriver\chromedriver_win32.zip
Received 781K...
Received 1568K...
Received 2352K...
Received 3136K...
Received 3299K total.
Extracting zip contents
Copying to target path C:\Workspace\Sites\vma\node_modules\chromedriver\lib\chromedriver
Done. ChromeDriver binary available at C:\Workspace\Sites\vma\node_modules\chromedriver\lib\chromedriver\chromedriver.exe

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\Workspace\Sites\vma\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1726 packages in 43.94s
```

**Comment by codemeall** [comment]
Im having error if I do npm run dev, 

 ERROR  Failed to compile with 1 errors

This dependency was not found:

* @/components/widgets/card/NameCard in ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/pages/widgets/Social.vue

To install it, you can run: npm install --save @/components/widgets/card/NameCard


But the above command npm install --save @/components/widgets/card/NameCard shows below error:

npm ERR! code ENOLOCAL
npm ERR! Could not install from "@/components/widgets/card/NameCard" as it does not contain a package.json file.

**Comment by TAnas0** [comment]
@AntoninSlejska Hi, thanks for answering.

I used `nvm` to switch to the same versions of `node` and `npm` that you are using, but I still face the same issue.

It is probably a problem in the version of a dependecy. Can I see your `package-lock.json` file? 

Here is mine:

[package-lock.txt](https://github.com/tookit/vue-material-admin/files/2199075/package-lock.txt)



Also note that `npm install` works fine for me and leads to exactly the same output as yours.
The problem is with the `npm run dev` command.

**Comment by tookit** [comment]
@TAnas0 my node: v9.8.0, NPM: 5.6.0
and can you check your `webpack.base.conf.js` , it seems that there is some issue about your `resolve` alias

`
  resolve: {
    extensions: ['.js', '.vue', '.json'],
    alias: {
      'vue$': 'vue/dist/vue.esm.js',
      '@': resolve('src'),
    }
  },
`

**Comment by ableabhijeet** [comment]
I am also facing NameCard dependency error.

```
This dependency was not found:

* @/components/widgets/card/NameCard in ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/pages/widgets/Social.vue

To install it, you can run: npm install --save @/components/widgets/card/NameCard
```

**Comment by AntoninSlejska** [comment]
Hi TAnas0,
here is my package-lock.json: [package-lock.txt](https://github.com/tookit/vue-material-admin/files/2200512/package-lock.txt)

**Comment by codemeall** [comment]
Still not sure what the problem is but I commented the script in "src/components/widgets/NameCard.vue" and the it works fine on server.

But the problem is the social page here "http://vma.isocked.com/#/widgets/social" is not working. Please let me know if anyone resolve the social page issue.

**Comment by AntoninSlejska** [comment]
Hi TAnas0,
I have updated node to  v8.11.3 and npm to 6.2.0 and the app works. I have only the problem with CORS. I suppose, that when I install the app at HTTPS, then the problem will be solved.

**Comment by AntoninSlejska** [comment]
I have tried npm update and outdated:

```
$ npm update
npm WARN deprecated istanbul-lib-hook@1.2.1: 1.2.0 should have been a major version bump

> chromedriver@2.40.0 install C:\Workspace\Sites\vma\node_modules\chromedriver
> node install.js

Downloading https://chromedriver.storage.googleapis.com/2.40/chromedriver_win32.zip
Saving to C:\Users\Slejska\AppData\Local\Temp\chromedriver\chromedriver_win32.zip
Received 781K...
Received 1568K...
Received 2352K...
Received 3136K...
Received 3376K total.
Extracting zip contents
Copying to target path C:\Workspace\Sites\vma\node_modules\chromedriver\lib\chromedriver
Done. ChromeDriver binary available at C:\Workspace\Sites\vma\node_modules\chromedriver\lib\chromedriver\chromedriver.exe
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN less-loader@4.1.0 requires a peer of less@^2.3.1 || ^3.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\sane\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ chromedriver@2.40.0
+ eslint-plugin-promise@3.8.0
+ babel-loader@7.1.5
+ babel-eslint@8.2.6
+ copy-webpack-plugin@4.5.2
+ eslint-plugin-import@2.13.0
+ jest@22.4.4
+ eslint-plugin-vue@4.7.0
+ postcss-loader@2.1.6
+ selenium-server@3.13.0
+ vue-loader@13.7.2
+ webpack-bundle-analyzer@2.13.1
+ webpack-merge@4.1.3
+ uglifyjs-webpack-plugin@1.2.7
+ vuetify@1.1.4
added 30 packages from 23 contributors, removed 13 packages, updated 59 packages, moved 2 packages and audited 30970 packages in 28.089s
found 164 vulnerabilities (72 low, 81 moderate, 8 high, 3 critical)
  run `npm audit fix` to fix them, or `npm audit` for details

$ npm outdated
Package                               Current     Wanted        Latest  Location
autoprefixer                            7.2.6      7.2.6         9.0.0  vue-material-admin
babel-jest                             21.2.0     21.2.0        23.4.0  vue-material-admin
babel-plugin-dynamic-import-node        1.2.0      1.2.0         2.0.0  vue-material-admin
cross-spawn                             5.1.0      5.1.0         6.0.5  vue-material-admin
css-loader                            0.28.11    0.28.11         1.0.0  vue-material-admin
eslint                                 4.19.1     4.19.1         5.1.0  vue-material-admin
eslint-config-standard                 10.2.1     10.2.1        11.0.0  vue-material-admin
eslint-friendly-formatter               3.0.0      3.0.0         4.0.1  vue-material-admin
eslint-loader                           1.9.0      1.9.0         2.0.0  vue-material-admin
eslint-plugin-node                      5.2.1      5.2.1         6.0.1  vue-material-admin
html-webpack-plugin                    2.30.1     2.30.1         3.2.0  vue-material-admin
jest                                   22.4.4     22.4.4        23.4.1  vue-material-admin
jest-serializer-vue                     0.3.0      0.3.0         2.0.2  vue-material-admin
json-schema-faker                   0.5.0-rc9  0.5.0-rc9    0.5.0-rc15  vue-material-admin
json-server                            0.12.2     0.12.2        0.14.0  vue-material-admin
optimize-css-assets-webpack-plugin      3.2.0      3.2.0         5.0.0  vue-material-admin
ora                                     1.4.0      1.4.0         2.1.0  vue-material-admin
shelljs                                 0.7.8      0.7.8         0.8.2  vue-material-admin
url-loader                              0.5.9      0.5.9         1.0.1  vue-material-admin
vee-validate                            2.0.9      2.0.9  2.1.0-beta.6  vue-material-admin
vue-jest                                1.4.0      1.4.0         2.6.0  vue-material-admin
vue-loader                             13.7.2     13.7.2        15.2.4  vue-material-admin
vue-style-loader                        3.1.2      3.1.2         4.1.0  vue-material-admin
webpack                                3.12.0     3.12.0        4.16.1  vue-material-admin
webpack-dev-server                     2.11.2     2.11.2         3.1.4  vue-material-admin
```

The app works, but I suppose, that some dependencies can be updated... But Vue is a new technology for me, so I'm not able to say, what should be done.

**Comment by TAnas0** [comment]
@tookit 
I still have the same error using node v9.8.0 and npm v5.6.0.
As to the alias `resolve` in `webpack.base.conf.js`, I have not changed it and is the same as yours.

@AntoninSlejska Thanks for providing your `package-lock.json`. I diffed them and they are the same.
I tried `npm update` and `npm outdated` which both give me exactly the same output as yours.

Thanks to @codemeall comment, we can pinpoint the cause of the error in the `NameCard` component.
The error is caused by a typo in the component's name: change name of component from `NameCard.Vue` to `NameCard.vue` and everything works as supposed to.

Still curious as to why this error was not experienced by everybody.

Thanks everybody. We solved a small problem that has probably cost this awesome repository some adopters.
